File: ControlFlow/python/EvenDigitSum.py
def even_digit_sum(number: int):
    if number < 0:
        return -1;
    return_sum = 0;
    while number > 0:
        if (number % 10) % 2 == 0: 
            return_sum += number % 10;
        number = int(number / 10);
    return return_sum;


print(even_digit_sum(10))
print(even_digit_sum(222))
print(even_digit_sum(-10))
print(even_digit_sum(2000004000002))

File: ControlFlow/java/EvenDigitSum.java
package ControlFlow.java;

public class EvenDigitSum {
    public static void main(String[] args) {
        System.out.println(getEvenDigitSum(123456789)); //→ should return 20 since 2 + 4 + 6 + 8 = 20

        System.out.println(getEvenDigitSum(252)); //→ should return 4 since 2 + 2 = 4

        System.out.println(getEvenDigitSum(-22)); //→ should return -1 since the number is negative
        
    }
    public static int getEvenDigitSum(int number){
        int returnSum = 0;
        if (number < 0){
            return -1;
        }
        while (number > 0){
            if ((number % 10) % 2 == 0){
                returnSum += number % 10;
            }
            number /= 10;
        }
        return returnSum;

    }
    
}
